<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>STM32Window</class>
 <widget class="QWidget" name="STM32Window">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>816</width>
    <height>692</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>STM32</string>
  </property>
  <property name="windowIcon">
   <iconset resource="../../../../resources/icons/buttons/buttonicons.qrc">
    <normaloff>:/icons/buttons/computer_edit.png</normaloff>:/icons/buttons/computer_edit.png</iconset>
  </property>
  <layout class="QGridLayout" name="gridLayout">
   <item row="1" column="0">
    <widget class="QGroupBox" name="groupBox_2">
     <property name="title">
      <string>Analog Input Gain</string>
     </property>
     <layout class="QGridLayout" name="gridLayout_4">
      <item row="2" column="2">
       <widget class="QGroupBox" name="groupBox_4">
        <property name="sizePolicy">
         <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <property name="title">
         <string>Receive gains</string>
        </property>
        <layout class="QGridLayout" name="gridLayout_5">
         <item row="3" column="1">
          <widget class="QLineEdit" name="sdadcGainConfigurationIndex1LineEdit">
           <property name="sizePolicy">
            <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
             <horstretch>0</horstretch>
             <verstretch>0</verstretch>
            </sizepolicy>
           </property>
           <property name="readOnly">
            <bool>true</bool>
           </property>
          </widget>
         </item>
         <item row="2" column="0">
          <widget class="QLabel" name="label_12">
           <property name="sizePolicy">
            <sizepolicy hsizetype="Fixed" vsizetype="Preferred">
             <horstretch>0</horstretch>
             <verstretch>0</verstretch>
            </sizepolicy>
           </property>
           <property name="text">
            <string>Gain for configuration index 0:</string>
           </property>
          </widget>
         </item>
         <item row="5" column="0" colspan="2">
          <widget class="QPushButton" name="receiveInputGainsPushButton">
           <property name="text">
            <string>Receive input gains</string>
           </property>
          </widget>
         </item>
         <item row="4" column="0">
          <widget class="QLabel" name="label_19">
           <property name="sizePolicy">
            <sizepolicy hsizetype="Fixed" vsizetype="Preferred">
             <horstretch>0</horstretch>
             <verstretch>0</verstretch>
            </sizepolicy>
           </property>
           <property name="text">
            <string>Gain for configuration index 2:</string>
           </property>
          </widget>
         </item>
         <item row="3" column="0">
          <widget class="QLabel" name="label_13">
           <property name="sizePolicy">
            <sizepolicy hsizetype="Fixed" vsizetype="Preferred">
             <horstretch>0</horstretch>
             <verstretch>0</verstretch>
            </sizepolicy>
           </property>
           <property name="text">
            <string>Gain for configuration index 1:</string>
           </property>
          </widget>
         </item>
         <item row="2" column="1">
          <widget class="QLineEdit" name="sdadcGainConfigurationIndex0LineEdit">
           <property name="sizePolicy">
            <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
             <horstretch>0</horstretch>
             <verstretch>0</verstretch>
            </sizepolicy>
           </property>
           <property name="readOnly">
            <bool>true</bool>
           </property>
          </widget>
         </item>
         <item row="4" column="1">
          <widget class="QLineEdit" name="sdadcGainConfigurationIndex2LineEdit">
           <property name="sizePolicy">
            <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
             <horstretch>0</horstretch>
             <verstretch>0</verstretch>
            </sizepolicy>
           </property>
           <property name="readOnly">
            <bool>true</bool>
           </property>
          </widget>
         </item>
        </layout>
       </widget>
      </item>
      <item row="0" column="2">
       <layout class="QVBoxLayout" name="verticalLayout">
        <item>
         <widget class="QLabel" name="label_7">
          <property name="text">
           <string>Configuration index 0 = ADC0, ADC1, ADC2, ADC9, DADC0, DADC1</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QLabel" name="label_8">
          <property name="text">
           <string>Configuration index 1 = ADC3, ADC4, ADC5, ADC10, DADC2, DADC3</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QLabel" name="label_14">
          <property name="text">
           <string>Configuration index 2 = ADC6, ADC7, ADC8, ADC11, DADC4</string>
          </property>
         </widget>
        </item>
       </layout>
      </item>
      <item row="2" column="1">
       <widget class="QGroupBox" name="groupBox_3">
        <property name="title">
         <string>Transmit gain</string>
        </property>
        <layout class="QGridLayout" name="gridLayout_3">
         <item row="2" column="0">
          <widget class="QLabel" name="label_11">
           <property name="text">
            <string>Gain</string>
           </property>
          </widget>
         </item>
         <item row="0" column="0">
          <widget class="QLabel" name="label_10">
           <property name="sizePolicy">
            <sizepolicy hsizetype="Fixed" vsizetype="Preferred">
             <horstretch>0</horstretch>
             <verstretch>0</verstretch>
            </sizepolicy>
           </property>
           <property name="text">
            <string>Configuration index:</string>
           </property>
          </widget>
         </item>
         <item row="0" column="1">
          <widget class="QComboBox" name="sdadcConfigurationIndexComboBox"/>
         </item>
         <item row="3" column="0" colspan="2">
          <widget class="QPushButton" name="sendInputGainPushButton">
           <property name="text">
            <string>Send input gain</string>
           </property>
          </widget>
         </item>
         <item row="2" column="1">
          <widget class="QComboBox" name="sdadcGainComboBox"/>
         </item>
        </layout>
       </widget>
      </item>
      <item row="0" column="1">
       <layout class="QVBoxLayout" name="verticalLayout_3">
        <item>
         <layout class="QHBoxLayout" name="horizontalLayout_2">
          <item>
           <widget class="QLabel" name="label_9">
            <property name="sizePolicy">
             <sizepolicy hsizetype="Fixed" vsizetype="Preferred">
              <horstretch>0</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
            <property name="text">
             <string>Analog peripheral:</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QComboBox" name="sdadcPeripheralComboBox">
            <property name="sizePolicy">
             <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
              <horstretch>0</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
           </widget>
          </item>
         </layout>
        </item>
        <item>
         <widget class="QLabel" name="label_4">
          <property name="text">
           <string>Peripheral 1 = ADC0 to ADC8</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QLabel" name="label_5">
          <property name="text">
           <string>Peripheral 2 = ADC9 to ADC11</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QLabel" name="label_6">
          <property name="text">
           <string>Peripheral 3 = DADC0 to DADC4</string>
          </property>
         </widget>
        </item>
       </layout>
      </item>
      <item row="3" column="1" colspan="2">
       <widget class="QGroupBox" name="groupBox_7">
        <property name="title">
         <string>Clock and alarm</string>
        </property>
        <layout class="QGridLayout" name="gridLayout_8">
         <item row="0" column="3" rowspan="4">
          <widget class="QGroupBox" name="groupBox_9">
           <property name="title">
            <string>Alarm B settings</string>
           </property>
           <layout class="QGridLayout" name="gridLayout_10">
            <item row="1" column="1">
             <widget class="QTimeEdit" name="alarmBTimeEdit"/>
            </item>
            <item row="0" column="1">
             <widget class="QComboBox" name="weekDayComboBox"/>
            </item>
            <item row="0" column="0">
             <widget class="QLabel" name="label_18">
              <property name="sizePolicy">
               <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
                <horstretch>0</horstretch>
                <verstretch>0</verstretch>
               </sizepolicy>
              </property>
              <property name="text">
               <string>Week day:</string>
              </property>
             </widget>
            </item>
            <item row="3" column="0">
             <widget class="QPushButton" name="receiveAlarmBPushButton">
              <property name="text">
               <string>Receive alarm B</string>
              </property>
             </widget>
            </item>
            <item row="1" column="0">
             <widget class="QLabel" name="label_21">
              <property name="text">
               <string>Time:</string>
              </property>
             </widget>
            </item>
            <item row="3" column="1">
             <widget class="QPushButton" name="sendAlarmBPushButton">
              <property name="text">
               <string>Send alarm B</string>
              </property>
             </widget>
            </item>
            <item row="2" column="0">
             <widget class="QCheckBox" name="alarmBEnabledCheckBox">
              <property name="text">
               <string>Alarm B enabled</string>
              </property>
             </widget>
            </item>
            <item row="2" column="1">
             <widget class="QCheckBox" name="alarmBActivatedCheckBox">
              <property name="enabled">
               <bool>true</bool>
              </property>
              <property name="text">
               <string>Alarm B activated</string>
              </property>
             </widget>
            </item>
           </layout>
          </widget>
         </item>
         <item row="0" column="2" rowspan="4">
          <widget class="QGroupBox" name="groupBox_8">
           <property name="title">
            <string>Alarm A settings</string>
           </property>
           <layout class="QGridLayout" name="gridLayout_9">
            <item row="1" column="0">
             <widget class="QLabel" name="label_20">
              <property name="text">
               <string>Time:</string>
              </property>
             </widget>
            </item>
            <item row="0" column="1">
             <widget class="QSpinBox" name="dateAlarmASpinBox">
              <property name="minimum">
               <number>1</number>
              </property>
              <property name="maximum">
               <number>31</number>
              </property>
             </widget>
            </item>
            <item row="0" column="0">
             <widget class="QLabel" name="label_17">
              <property name="sizePolicy">
               <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
                <horstretch>0</horstretch>
                <verstretch>0</verstretch>
               </sizepolicy>
              </property>
              <property name="text">
               <string>Date:</string>
              </property>
             </widget>
            </item>
            <item row="1" column="1">
             <widget class="QTimeEdit" name="alarmATimeEdit"/>
            </item>
            <item row="3" column="1">
             <widget class="QPushButton" name="sendAlarmAPushButton">
              <property name="text">
               <string>Send alarm A</string>
              </property>
             </widget>
            </item>
            <item row="3" column="0">
             <widget class="QPushButton" name="receiveAlarmAPushButton">
              <property name="text">
               <string>Receive alarm A</string>
              </property>
             </widget>
            </item>
            <item row="2" column="0">
             <widget class="QCheckBox" name="alarmAEnabledCheckBox">
              <property name="text">
               <string>Alarm A enabled</string>
              </property>
             </widget>
            </item>
            <item row="2" column="1">
             <widget class="QCheckBox" name="alarmAActivatedCheckBox">
              <property name="enabled">
               <bool>true</bool>
              </property>
              <property name="text">
               <string>Alarm A activated</string>
              </property>
             </widget>
            </item>
           </layout>
          </widget>
         </item>
         <item row="0" column="1" rowspan="4">
          <widget class="QGroupBox" name="groupBox_10">
           <property name="title">
            <string>Clock</string>
           </property>
           <layout class="QGridLayout" name="gridLayout_11">
            <item row="1" column="0">
             <widget class="QPushButton" name="receiveDateTimePushButton">
              <property name="text">
               <string>Receive date time</string>
              </property>
             </widget>
            </item>
            <item row="0" column="0">
             <widget class="QDateTimeEdit" name="dateTimeEdit">
              <property name="date">
               <date>
                <year>2000</year>
                <month>1</month>
                <day>1</day>
               </date>
              </property>
              <property name="time">
               <time>
                <hour>0</hour>
                <minute>0</minute>
                <second>0</second>
               </time>
              </property>
              <property name="maximumDateTime">
               <datetime>
                <hour>23</hour>
                <minute>59</minute>
                <second>59</second>
                <year>2099</year>
                <month>12</month>
                <day>31</day>
               </datetime>
              </property>
              <property name="minimumDateTime">
               <datetime>
                <hour>0</hour>
                <minute>0</minute>
                <second>0</second>
                <year>2000</year>
                <month>1</month>
                <day>1</day>
               </datetime>
              </property>
             </widget>
            </item>
            <item row="2" column="0">
             <widget class="QPushButton" name="sendDateTimePushButton">
              <property name="text">
               <string>Send date time</string>
              </property>
             </widget>
            </item>
           </layout>
          </widget>
         </item>
        </layout>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item row="0" column="0">
    <widget class="QGroupBox" name="groupBox">
     <property name="title">
      <string>PWM Prescaler</string>
     </property>
     <layout class="QGridLayout" name="gridLayout_7">
      <item row="0" column="0">
       <widget class="QGroupBox" name="groupBox_5">
        <property name="title">
         <string>Transmit prescaler</string>
        </property>
        <layout class="QGridLayout" name="gridLayout_2">
         <item row="1" column="0">
          <widget class="QLabel" name="pwmPrescalerLabel">
           <property name="sizePolicy">
            <sizepolicy hsizetype="Fixed" vsizetype="Preferred">
             <horstretch>0</horstretch>
             <verstretch>0</verstretch>
            </sizepolicy>
           </property>
           <property name="text">
            <string>PWM prescaler:</string>
           </property>
          </widget>
         </item>
         <item row="2" column="0" colspan="2">
          <widget class="QPushButton" name="sendPrescalerPushButton">
           <property name="text">
            <string>Send prescaler</string>
           </property>
          </widget>
         </item>
         <item row="1" column="1">
          <widget class="QSpinBox" name="pwmPrescalerSpinBox">
           <property name="maximum">
            <number>65535</number>
           </property>
          </widget>
         </item>
         <item row="0" column="0">
          <widget class="QLabel" name="label">
           <property name="sizePolicy">
            <sizepolicy hsizetype="Fixed" vsizetype="Preferred">
             <horstretch>0</horstretch>
             <verstretch>0</verstretch>
            </sizepolicy>
           </property>
           <property name="text">
            <string>PWM peripheral:</string>
           </property>
          </widget>
         </item>
         <item row="0" column="1">
          <widget class="QComboBox" name="pwmPeripheralComboBox"/>
         </item>
        </layout>
       </widget>
      </item>
      <item row="0" column="2">
       <widget class="QGroupBox" name="groupBox_6">
        <property name="title">
         <string>Receive prescalers</string>
        </property>
        <layout class="QGridLayout" name="gridLayout_6">
         <item row="0" column="0">
          <widget class="QLabel" name="label_3">
           <property name="sizePolicy">
            <sizepolicy hsizetype="Fixed" vsizetype="Preferred">
             <horstretch>0</horstretch>
             <verstretch>0</verstretch>
            </sizepolicy>
           </property>
           <property name="text">
            <string>PWM 0-3 prescaler:</string>
           </property>
          </widget>
         </item>
         <item row="0" column="1">
          <widget class="QSpinBox" name="pwm_0_3_prescalerSpinBox">
           <property name="readOnly">
            <bool>true</bool>
           </property>
           <property name="buttonSymbols">
            <enum>QAbstractSpinBox::NoButtons</enum>
           </property>
           <property name="maximum">
            <number>65535</number>
           </property>
          </widget>
         </item>
         <item row="1" column="0">
          <widget class="QLabel" name="label_2">
           <property name="sizePolicy">
            <sizepolicy hsizetype="Fixed" vsizetype="Preferred">
             <horstretch>0</horstretch>
             <verstretch>0</verstretch>
            </sizepolicy>
           </property>
           <property name="text">
            <string>PWM 4-7 prescaler:</string>
           </property>
          </widget>
         </item>
         <item row="1" column="1">
          <widget class="QSpinBox" name="pwm_4_7_prescalerSpinBox">
           <property name="readOnly">
            <bool>true</bool>
           </property>
           <property name="buttonSymbols">
            <enum>QAbstractSpinBox::NoButtons</enum>
           </property>
           <property name="maximum">
            <number>65535</number>
           </property>
          </widget>
         </item>
         <item row="2" column="0" colspan="2">
          <widget class="QPushButton" name="receivePrescalersPushButton">
           <property name="text">
            <string>Receive prescalers</string>
           </property>
          </widget>
         </item>
        </layout>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
  </layout>
 </widget>
 <resources>
  <include location="../../../../resources/icons/buttons/buttonicons.qrc"/>
  <include location="../../../../resources/icons/buttons/buttonicons.qrc"/>
  <include location="../../../../resources/icons/buttons/buttonicons.qrc"/>
 </resources>
 <connections/>
</ui>
